packages/configs/eslint-common-react.json

```diff
@@ -1,9 +1,5 @@
 {
-    "extends": [
-        "airbnb",
-        "./eslint-common.json",
-        "prettier/react"
-    ],
+    "extends": ["airbnb", "./eslint-common.json", "plugin:react/recommended", "prettier/react"],
```

Comment:
I realized that this way we are actually overwriting `eslint-plugin-react` rules set in `airbnb` config, which I'm not sure is the right approach if we are sticking to AirBnB...

I see that so far we used `plugin:react/recommended` always before `eslint-common-react` in Stage and Composer, and because of that we didn't get notifications about `react/jsx-key` (https://github.com/airbnb/javascript/blob/fd77bbebb77362ddecfef7aba3bf6abf7bdd81f2/packages/eslint-config-airbnb/rules/react.js#L95C28-L95C28). In Components we have the other way (first airbnb, then recommended settings). It's a mess, right, but before making any changes I'd like to learn more.

What are the differences between AirBnB and `plugin:react/recommended`? Maybe we should end up with adding `plugin:react/recommended` before `airbnb`? Maybe we could just override `react/jsx-key` rule? Maybe `plugin:react/recommended` and `airbnb` are similar (in terms of `eslint-plugin-react`) and we should be good with your approach? These are questions worth to ask first.

Reply:
Good point!
I'll try to clarify those questions 😁 

Airbnb is extending `eslint-plugin-react` config in a way shown in [this file](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/rules/react.js)
As you've pointed out [this config](https://github.com/airbnb/javascript/blob/fd77bbebb77362ddecfef7aba3bf6abf7bdd81f2/packages/eslint-config-airbnb/rules/react.js#L95C28-L95C28) is disabling `react/jsx-key` rule

As there are much more react rules adjusted by airbnb config I think that listing all turned off, may help us in making the decision related to final eslint configuration for react 💪

Reply:
List of disabled rules:
|  name of disabled eslint rule | short rule description | airbnb comment | 
|---|---|---|
|  react/display-name  |  Prevent missing displayName in a React component definition | |
| react/forbid-dom-props  |  Forbid certain props on DOM Nodes | |
|  react/jsx-handler-names  |  Enforce event handler naming conventions in JSX | |
|  react/jsx-key |  Validate JSX has key prop when in array or iterator | Turned off because it has too many false positives |
|  react/jsx-no-literals |  Prevent usage of unwrapped JSX strings |   | 
|  react/sort-prop-types |  Enforce propTypes declarations alphabetical sorting |   | 
|  react/jsx-sort-prop-types  | Deprecated in favor of react/jsx-sort-props |   | 
|  react/jsx-sort-props |  Enforce props alphabetical sorting|   | 
|  react/jsx-sort-default-props | Enforce defaultProps declarations alphabetical sorting  |   | 
|  react/no-did-mount-set-state` |  Prevent usage of setState in componentDidMount |  this is necessary for server-rendering | 
|  react/no-direct-mutation-state |  Prevent direct mutation of this.state |   | 
|  react/no-multi-comp |  Prevent multiple component definition per file |   | 
|  react/no-set-state |  Prevent usage of setState |   | 
|  react/require-optimization |   require a shouldComponentUpdate method, or PureRenderMixin |   | 
|  react/forbid-component-props  |  Forbid certain props on Components  |   | 
|  react/forbid-elements |   Forbid certain elements |   | 
|  react/jsx-space-before-closing |  Enforce spaces before the closing bracket of self-closing JSX elements  |  Deprecated in favor of jsx-tag-spacing | 
| react/boolean-prop-naming  |  Enforces consistent naming for boolean props |   | 
|  react/jsx-child-element-spacing |  Ensures inline tags are not rendered without spaces between them |   | 
|  react/jsx-max-depth  | Validate JSX maximum depth  |   | 
|  react/no-unsafe |  Prevent usage of UNSAFE_ methods |   | 
|  react/prefer-read-only-props | Enforce that props are read-only  |   | 
|  react/no-adjacent-inline-elements |  Prevent adjacent inline elements not separated by whitespace  |   | 
|  react/jsx-newline |  Enforce a new line after jsx elements and expressions |   | 
|  react/hook-use-state |   Ensure destructuring and symmetric naming of useState hook value and setter variables |   | 
|  react/iframe-missing-sandbox |  Enforce sandbox attribute on iframe elements |  TODO: semver-major, enable | 
|  react/jsx-no-leaked-render  | Prevent problematic leaked values from being rendered  |  TODO: semver-major, enable | 
|  react/no-object-type-as-default-prop|   |   TODO: semver-major, enable | 
|  react/sort-default-props  |   | TODO: semver-major, enable?  |

Reply:
Thanks for checking the differences, indeed they seem to be acceptable from our perspective, so I agree with your approach.

Reply:
Great to hear 🎉 
If you'd thought about any more questions, feel free to mention them :)